test/framework/netns/netns.go

```diff
@@ -0,0 +1,526 @@
+package netns
+
+import (
+	"fmt"
+	"math"
+	"net"
+	"runtime"
+	"strings"
+	"syscall"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
```

Comment:
The docs for this are interesting. It looks like an understanding of golang threading model is necessary to be sure everything works as expected. 

I've found this documentation of the scheduler to be very useful, and the right level of detail -

https://www.ardanlabs.com/blog/2018/08/scheduling-in-go-part2.html

Reply:
Nice! We should add this link somewhere

Reply:
added this link in the comment of netns.go